### PR TITLE
Fixes for SerialWriter and SerialReader

### PR DESCRIPTION
**Pull request by veggiemike**
Hey, Pablo!  Here's my next batch of fixes!  Very similar to the ones you just reviewed for UDPWriter, this time I was working on serial data.  I'm not even using serial here at the moment, but test suite failures caught my eye while I was compare-n-contrasting working on the UDP stuff.

I also included a commit that makes the 2 trivial changes you mentioned reviewing PR #345.

### CODE REVIEW

**Review by davidpablocohn — APPROVED** [review]
Two little typos, but other than that, this also looks great - thank you!

I've approved the pull, but rather than merging right now, I'll stand by for a bit and work on the remaining review in case you have a chance to address the typos.

File: logger/writers/test_serial_writer.py [hunk 1]
```diff
@@ -155,55 +181,47 @@ def test_write(self):
...
-            reader = SerialReader(port)
-            for line in SAMPLE_DATA.split('\n'):
-                result = reader.read()
-                logging.info('data: %s, read: %s', line, result)
-                self.assertEqual(line, result)
-
         def _run_writer(in_port):
             writer = SerialWriter(port=in_port)
             for line in SAMPLE_DATA.split('\n'):
                 logging.info('wrote: %s', line)
```
[the remark this answers is not in the document]
Comment (veggiemike) [reply logger/writers/test_serial_writer.py:187]: Huh, you're absolutely right.  I'll move the log message to after the write in both places it's backwards.

File: logger/readers/serial_reader.py [hunk 2]
```diff
@@ -93,23 +92,36 @@ def __init__(self,  port, baudrate=9600, bytesize=8, parity='N',
...
+            eol = self._encode_str(eol, unescape=True)
         self.eol = eol
 
     ############################
     def read(self):
         try:
             if self.eol:
-                record = self.serial.read_until(terminator=self.eol, size=self.max_bytes)
+                record = self.serial.read_until(expected=self.eol, size=self.max_bytes)
+                # read_until()'s record includes a trialing 'eol', strip it off
```
[the remark this answers is not in the document]
Comment (veggiemike) [reply logger/readers/serial_reader.py:109]: Gah, typing on a bouncing boat.  ;-)  I'll fix this for you, too.

**Comment by veggiemike** [comment]
I'll push another commit out to fix those little things.

**Pull request merged**